app/controllers/users/verify_account_controller.rb

```diff
@@ -8,9 +8,9 @@ def index
 
     def create
       @verify_account_form = build_verify_account_form
+
       if @verify_account_form.submit
-        flash[:success] = t('account.index.verification.success')
-        redirect_to account_path
+        redirect_to sign_up_completed_url
```

Comment:
Also, this assumes that the initial SP request is still in the session, but in all likelihood, it will not be, since by the time the user receives the letter, the session will have expired, so when they sign back in, `session[:sp]` will not be present, and redirecting them to `sign_up_completed_url` will take them to the account page.